pydarn/io/borealis/borealis_formats.py

```diff
@@ -246,27 +353,428 @@ def array_single_element_types(cls):
+        the site data with all arrays flattened as desired for
+        storing in the site format.
+        """
+        for key in list(records.keys()):
+            for field in ['main_acfs', 'intf_acfs', 'xcfs']:
+                records[key][field] = records[key][field].flatten()
+
+        return records
+
+    @classmethod
+    def single_element_types(cls):
+        return {
+            # Identifies the version of Borealis that made this data. Necessary
+            # for all versions.
+            "borealis_git_hash": np.unicode_,
+            # Number used to identify experiment.
+            "experiment_id": np.int64,
+            # Name of the experiment file.
+            "experiment_name": np.unicode_,
+            # Comment about the whole experiment
+            "experiment_comment": np.unicode_,
+            # Additional text comment that describes the slice.
+            "slice_comment": np.unicode_,
+            # Number of slices in the experiment at this integration time.
+            "num_slices": np.int64,
+            # Three letter radar identifier.
+            "station": np.unicode_,
+            # Number of sampling periods in the integration time.
+            "num_sequences": np.int64,
+            # range gate separation (equivalent distance between samples), km.
+            "range_sep": np.float32,
+            # Round trip time of flight to first range in microseconds.
+            "first_range_rtt": np.float32,
+            # Distance to first range in km.
+            "first_range": np.float32,
+            # Sampling rate of the samples being written to file in Hz.
+            "rx_sample_rate": np.float64,
+            # Designates if the record is the first in a scan.
+            "scan_start_marker": np.bool_,
+            # Integration time in seconds.
+            "int_time": np.float32,
+            # Length of the pulse in microseconds.
+            "tx_pulse_len": np.uint32,
+            # The minimum spacing between pulses, spacing between pulses is
+            # always a multiple of this in microseconds.
+            "tau_spacing": np.uint32,
+            # Number of main array antennas.
+            "main_antenna_count": np.uint32,
+            # Number of interferometer array antennas.
+            "intf_antenna_count": np.uint32,
+            # The frequency used for this experiment slice in kHz.
+            "freq": np.uint32,
+            # str denoting C data type of the samples included in the data
+            # array, such as 'complex float'.
+            "samples_data_type": np.unicode_,
+            # data normalization factor determined by the filter scaling in the
+            # decimation scheme.
+            "data_normalization_factor": np.float64,
+            # number of beams calculated for the integration time.
+            "num_beams": np.uint32
+            }
+
+    @classmethod
+    def array_dtypes(cls):
+        return {
```

Comment:
Again could this be an attribute of class attribute so you can call like a variable?